### ISSUE TITLE
Allow override of modules

### ISSUE THREAD

**Issue by geekingfrog** [geekingfrog/zeninjector#4]
Currently, calling `container.register` with an already used name will throw an error. Allow an option (on the container object or on `register` ?) to allow this behavior.
This is useful for testing, when you want to only change one dependency.

**Comment by TomKaltz** [comment]
+1

**Comment by geekingfrog** [comment]
@TomKaltz can you tell me a bit more about your usecase? I realized just adding an option to override a dependency would also require to re-evaluate all other modules which depends on the overridden module, making it a much bigger change than expected.

**Comment by TomKaltz** [comment]
Yeah you're probably, right.  Changing out objects during execution isn't
the best idea.  I was thinking about testing scenarios and passing in mocks.

Different singletons should be registered in production and testing.  The
developer will have to figure out how to separate this wiring.

**Comment by geekingfrog** [comment]
What I'm is something along the line:

``` javascript
// @autoinject
module.exports.myModule = function(fs) {
...
}
```

``` javascript
var moduleToTest;
beforeEach(function() {
  var fakeFs = ...;
  moduleToTest = constructor(fakeFs);
})
```

And if you need access to the other dependencies (not mocked), you can always use `container.inject`.

**Comment by Pajk** [comment]
@geekingfrog I also needed the option to mock some modules in tests and [here is my simple solution](https://github.com/Pajk/zeninjector/commit/619d2aae6e00d9334977555550b4c8f7c300fa8c). 

You can use `scan` to load all mocks and then all app modules this way:

```
var container = new ZenInjector({
    mock_modules: {
        upload: true
    }
})
yield container.scan(__dirname + "/test/mock/**/*.js")
yield container.scan(__dirname + "/src/**/*.js")
```
